Source/cmNinjaTargetGenerator.cxx

```diff
@@ -340,12 +340,42 @@ ::WriteCompileRule(const std::string& lang)
   vars.ObjectFileDir = "$OBJECT_FILE_DIR";
 
   cmMakefile* mf = this->GetMakefile();
+  
+  const char* forceRspFile = "CMAKE_NINJA_FORCE_RESPONSE_FILE";
+  bool useResponseFile = mf->IsDefinitionSet(forceRspFile) ||
+                         cmSystemTools::GetEnv(forceRspFile) != 0;
+
+  std::string flags = "$FLAGS";
+  std::string rspfile;
+  std::string rspcontent;
+  std::string responseFlag;
+
+  if (useResponseFile) {
+    std::string cmakeVarLang = "CMAKE_";
+    cmakeVarLang += lang;
+
+    // build response file name
+    std::string cmakeLinkVar =  cmakeVarLang + "_RESPONSE_FILE_LINK_FLAG";
+    const char * flag = GetMakefile()->GetDefinition(cmakeLinkVar);
+    if(flag) {
+      responseFlag = flag;
+    } else {
+      responseFlag = "@";
+    }
```

Comment:
We should be able to just hard-code use of `@` here because the only reason we have `_RESPONSE_FILE_LINK_FLAG` is because we do not know whether to use `@` or `-Wl,@` when invoking a linker.  For the compiler we are always passing the response file directly.